config/config.go

```diff
@@ -135,6 +139,13 @@ func (s *HTTPProbe) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := s.HTTPClientConfig.Validate(); err != nil {
 		return err
 	}
+	if s.BodyPath != "" {
```

Comment:
This should be opened and read in the request. It should also be an error for both to be specified.

Reply:
Updated to be opened within `ProbeHTTP`. I left the current `os.Open` call in `config.go` in order to fail fast on bad configs. This is different from `BearerToken`/`BearerTokenFile` (https://github.com/prometheus/common/blob/master/config/http_config.go#L81).

Let me know if I should drop opening the file during config validation.